images/nginx-dbt-platform/entrypoint.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Either PRIV_PATH_LIST or PUB_PATH_LIST VARs can be set, not both.
+# If neither is set, the default is to make / public
+# To enable IP filter set PRIV_PATH_LIST: '/'
+if ! [ -z ${PRIV_PATH_LIST+x} ]; then
+  PUBLIC_PATHS=""
+elif [ -z ${PUB_PATH_LIST+x} ]; then
+  echo -e "    location / {\n\tproxy_pass http://upstream_server_public;\n\tproxy_set_header Host \$host;\n\tproxy_set_header x-forwarded-for \$proxy_add_x_forwarded_for;\n\tproxy_set_header X-Forwarded-Prefix /;\n    }\n" >> public_paths.txt
```

Comment:
I would be inclined to extract these out to parameterised method to reduce duplication and make it easier to see what's actually different between them.

Reply:
updated with function.